src/coreclr/jit/importer.cpp

```diff
@@ -4517,106 +4517,84 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
             {
                 assert(varTypeIsFloating(callType));
 
-                if (sig->numArgs == 2 && (impStackTop().val->IsCnsFltOrDbl() || impStackTop(1).val->IsCnsFltOrDbl()))
+                if ((impStackTop().val->IsCnsFltOrDbl() || impStackTop(1).val->IsCnsFltOrDbl()))
                 {
                     GenTree* op2 = impPopStack().val;
                     GenTree* op1 = impPopStack().val;
 
-                    unsigned int insSimdSize = 16; //(callType == TYP_DOUBLE) ? 32 : 16;
-
                     // 1. NaN is propagated if either input is NaN
                     if (op1->IsFloatNaN() || op2->IsFloatNaN())
                     {
                         GenTree* ret = op1->IsFloatNaN() ? op1 : op2;
-                        retNode =
-                            gtNewSimdHWIntrinsicNode(callType, ret, NI_Vector128_ToScalar, callJitType, insSimdSize);
+                        retNode      = gtNewSimdHWIntrinsicNode(callType, ret, NI_Vector128_ToScalar, callJitType, 16);
```

Comment:
This can just be `retNode = op1->IsFloatNaN() ? op1 : op2;` as we already have the correct scalar value.